src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -531,5 +531,22 @@ private static bool IsHex(char ch)
                 // Check if less than 6 representing chars 'a' - 'f'
                 || (uint)((ch | 32) - 'a') < 6u;
         }
+
+        public static AltSvcHeader? GetEndpointAltSvc(System.Net.IPEndPoint endpoint, HttpProtocols protocols)
+        {
+            var hasHttp1OrHttp2 = protocols.HasFlag(HttpProtocols.Http1) || protocols.HasFlag(HttpProtocols.Http2);
+            var hasHttp3 = protocols.HasFlag(HttpProtocols.Http3);
+
+            if (hasHttp1OrHttp2 && hasHttp3)
+            {
+                var text = $"h3=\":{endpoint.Port}\"; ma=84600";
```

Comment:
https://datatracker.ietf.org/doc/html/rfc7838#section-3.1

>    When an alternative service is advertised using Alt-Svc, it is
   considered fresh for 24 hours from generation of the message.  This
   can be modified with the "ma" (max-age) parameter.

84600 = 23.5 hours? Why this value? And why specify `ma` at all if the default is 24h?

Reply:
I copied the value that was in .NET 5. I don't know why that was chosen.

Google uses `2592000`, which is 30 days.

Reply:
Facebook uses `3600`. 1 hour.

Reply:
Probably better to just remove it and use the default for now.

Reply:
Cloudflare is `2592000`

Reply:
Fastly is `86400`, 24 hours.

Reply:
I'm going to use 86400. I know it is the default, but every deployed website I've visited that uses h3 specifies a value. No one relies on the default. Even websites that have the default value (24 hours) specify ma=86400.

Reply:
This is making an argument for a `TimeSpan? AltSvcMaxAge` property.

Reply:
> 84600 = 23.5 hours? Why this value? And why specify `ma` at all if the default is 24h?
Maybe it was a typo, and 6 and 4 were reversed 😄

Reply:
> This is making an argument for a `TimeSpan? AltSvcMaxAge` property.

IMO wait and see if anyone asks for it.